common/tmx_level.go

```diff
@@ -87,15 +84,10 @@ func (t ByFirstgid) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
 func (t ByFirstgid) Less(i, j int) bool { return t[i].Firstgid < t[j].Firstgid }
 
 // MUST BE base64 ENCODED and COMPRESSED WITH zlib!
-func createLevelFromTmx(r Resource) (*Level, error) {
+func createLevelFromTmx(tmx string) (*Level, error) {
 	tlvl := &TMXLevel{}
 	lvl := &Level{}
 
-	tmx, err := readTmx(r.url)
-	if err != nil {
-		return lvl, err
-	}
-
 	if err := xml.Unmarshal([]byte(tmx), &tlvl); err != nil {
 		fmt.Printf("Error unmarshalling XML: %v", err)
```

Comment:
But it shouldn't log / print if it returns an error. The receiving end can parse / print / panic the error has he pleases.